liboptparse.so has mysterious unresolved jansson symbols on focal

Problem: as noted in flux-framework/flux-pmix#102, flux-pmix stopped working in the `focal` image.  The problem is that a test executable cannot be linked:
```
 garlick@system76-pc:/usr/src/t/src$ make V=1 barrier
/bin/bash ../../libtool  --tag=CC   --mode=link gcc -Wall -Werror -Wno-strict-aliasing -Wno-error=deprecated-declarations  -g -O2   -o barrier barrier.o ./libtestutil.la ../../src/common/libutil/libutil.la -lflux-optparse -lflux-idset -Wl,-rpath -Wl,/usr/lib -Wl,--enable-new-dtags -lmpi -Wl,-rpath -Wl,/usr/lib -Wl,--enable-new-dtags -lpmix 
libtool: link: gcc -Wall -Werror -Wno-strict-aliasing -Wno-error=deprecated-declarations -g -O2 -o barrier barrier.o -Wl,-rpath -Wl,/usr/lib -Wl,--enable-new-dtags -Wl,-rpath -Wl,/usr/lib -Wl,--enable-new-dtags  ./.libs/libtestutil.a ../../src/common/libutil/.libs/libutil.a /usr/lib/libflux-optparse.so /usr/lib/libflux-idset.so /usr/lib/libmpi.so /usr/lib/libpmix.so -pthread
/usr/bin/ld: /usr/lib/libflux-optparse.so: undefined reference to `json_object_iter_value'
/usr/bin/ld: /usr/lib/libflux-optparse.so: undefined reference to `json_object_key_to_iter'
/usr/bin/ld: /usr/lib/libflux-optparse.so: undefined reference to `json_object_set_new_nocheck'
/usr/bin/ld: /usr/lib/libflux-optparse.so: undefined reference to `json_object_iter_next'
/usr/bin/ld: /usr/lib/libflux-optparse.so: undefined reference to `json_object_iter_key'
/usr/bin/ld: /usr/lib/libflux-optparse.so: undefined reference to `json_object_get'
/usr/bin/ld: /usr/lib/libflux-optparse.so: undefined reference to `json_object_iter'
collect2: error: ld returned 1 exit status
make: *** [Makefile:516: barrier] Error 1
```
Apparently the installed libflux-optparse.so (0.69.0-55-g7cd67057e) has unresolved jansson symbols in it.

This is not a problem in the `el8` and `jammy` builders.

The last action in the flux-pmix repo was a succeeding one 10 months ago, so that does not help narrow things down.
I *thought* the the pmix check in the flux-core CI had been failing for some time, but looking through past actions, that doesn't seem to be the case!  Also, that check is done in `ubuntu-latest`.

A concern is that this problem emerged after the linkage changes for macos portability last week.  It will require more investigation.

--- a/configure.ac
+++ b/configure.ac
@@ -166,6 +166,20 @@ AC_PATH_PROGS(SSH, [rsh ssh], [/usr/bin/rsh])
 AC_DEFINE_UNQUOTED([PATH_SSH], "$SSH",
           [Define remote shell program to be used by the ssh:// connector])
 
+# macos linker doesn't support --gc-sections
+saved_LDFLAGS="$LDFLAGS"
+LDFLAGS="-Wl,--gc-sections $LDFLAGS"
+AC_MSG_CHECKING([whether ld supports --gc-sections])
+AC_LINK_IFELSE([AC_LANG_PROGRAM([])], [
+  AC_MSG_RESULT([yes])
+  ld_gc_sections="-Wl,--gc-sections"
+  AS_VAR_SET(ld_gc_sections, $ld_gc_sections)
+  AC_SUBST(ld_gc_sections)
+  ],[
+  AC_MSG_RESULT([no])
+])
+LDFLAGS=$saved_LDFLAGS
+
 LT_INIT
 AC_PROG_AWK
 

